src/test/java/puzzles/nurikabe/rules/BlackOrWhiteCaseRuleTest.java

```diff
@@ -25,27 +31,45 @@ public static void setUp() {
         nurikabe = new Nurikabe();
     }
 
+    /**
+     * Tests the Black Or White case rule by ensuring that it results in two  children, that contain a modified cell that is either black or white
+     */
     @Test
     public void TooFewSpacesContradictionRule_TwoSurroundBlackTest() throws InvalidFileFormatException {
-        TestUtilities.importTestBoard("puzzles/nurikabe/rules/TooFewSpacesContradictionRule/TwoSurroundBlack", nurikabe);
+        TestUtilities.importTestBoard("puzzles/nurikabe/rules/BlackOrWhiteCaseRule/SimpleBlackOrWhite", nurikabe);
         TreeNode rootNode = nurikabe.getTree().getRootNode();
         TreeTransition transition = rootNode.getChildren().get(0);
         transition.setRule(RULE);
 
-        Assert.assertNull(RULE.checkContradiction((NurikabeBoard) transition.getBoard()));
-
         NurikabeBoard board = (NurikabeBoard) transition.getBoard();
-        Point location = new Point(1, 1);
-        for (int i = 0; i < board.getHeight(); i++) {
-            for (int k = 0; k < board.getWidth(); k++) {
-                Point point = new Point(k, i);
-                if (point.equals(location)) {
-                    Assert.assertNull(RULE.checkRuleAt(transition, board.getCell(k, i)));
-                }
-                else {
-                    Assert.assertNotNull(RULE.checkRuleAt(transition, board.getCell(k, i)));
+        NurikabeCell cell = board.getCell(0,0);
+        ArrayList<Board> cases = RULE.getCases(board,cell);
+
+        Assert.assertEquals(2,cases.size());
+
+        NurikabeBoard caseBoard = (NurikabeBoard) cases.get(0);
+        NurikabeBoard caseBoard2 = (NurikabeBoard) cases.get(1);
+
+        Integer v1 = new Integer(0);
+        Integer v2 = new Integer(-1);
+
+        Assert.assertTrue((caseBoard.getCell(0,0).getData().equals(v1) || caseBoard.getCell(0,0).equals(v2)) &&
+                (caseBoard2.getCell(0,0).getData().equals(v1) || caseBoard2.getCell(0,0).getData().equals(v2)));
+        Assert.assertFalse(caseBoard.getCell(0,0).getData().equals(caseBoard2.getCell(0,0).getData()));
```

Comment:
Instead of using Integer objects, use NurikabeCell.getType(). This will return a NurikabeType that you can do the equality checks with. It will make the code more readable.